src/types/types.ts

```diff
@@ -805,6 +805,165 @@ export interface GetFloorPriceResponse {
+  buyerAddress?: string;
+
+  /** The address of the NFT seller to filter sales by. */
+  sellerAddress?: string;
+
+  /** Filter by whether the buyer or seller was the taker in the NFT trade. */
+  taker?: NftTakerType;
+
+  /** The maximum number of NFT sales to return. */
+  limit?: number;
+
+  /** Key for pagination to use to fetch results from the next page if available. */
+  pageKey?: string;
+}
+
+/**
+ * Alternative optional parameters object for the {@link getNftSales} endpoint
+ * that allows filtering results by contractAddress.
+ *
+ * This interface is used to filter the NFT sales data.
+ *
+ * @public
+ */
+export interface GetNftSalesByContractAddress extends GetNftSales {
+  /** The contract address of a NFT collection to filter sales by. */
+  contractAddress: string;
+
+  /** The token ID of an NFT within the specified contractAddress to filter sales by. */
+  tokenId?: BigNumberish;
+}
+
+/**
+ * The response for the {@link NftNamespace.getNftSales} method.
+ *
+ * @public
+ */
+export interface GetNftSalesResponse {
+  /** The page key to use to fetch the next page if more results are available. */
+  pageKey?: string;
+
+  /** List of NFT sales that match the query */
+  nftSales: NftSale[];
+}
+
+/** A single NFT sale data. */
+export interface NftSale {
+  /** The marketplace the sale took place on. */
+  marketplace: NftSaleMarketplace;
+
+  /** The NFT contract address. */
+  contractAddress: string;
+
+  /** The decimal token ID of the NFT being sold. */
+  tokenId: string;
+
+  /** The number of tokens sold in the sale as a decimal integer string. */
+  quantity: string;
+
+  /** The address of the buyer in the NFT sale. */
+  buyerAddress: string;
+
+  /** The address of the seller in the NFT sale. */
+  sellerAddress: string;
+
+  /** Whether the price taker in the trade was the buyer or the seller. */
+  taker: NftTakerType;
+
+  /** The payment from buyer to the seller. */
+  sellerFee?: NftSaleFeeData;
+
+  /** The payment from buyer to the marketplace. */
+  marketplaceFee?: NftSaleFeeData;
+
+  /** The payment from buyer to the royalty address of the NFT collection. */
+  royaltyFee?: NftSaleFeeData;
+
+  /** The block number the NFT sale took place in. */
+  blockNumber?: number;
+
+  /** The log number of the sale event emitted within the block. */
+  logIndex: number;
+
+  /** The index of the token within the bundle of NFTs sold in the sale. */
+  bundleIndex: number;
+
+  /** The transactionHash of the NFT sale. */
+  transactionHash: string;
+}
+
+/**
+ * Fee detail for an NFT sale.
+ *
+ * @public
+ */
+export interface NftSaleFeeData {
+  /** The fee payment amount as a decimal integer string. */
+  amount: string;
+
+  /** The symbol of the token used for the payment. */
+  symbol: string;
+
+  /** The number of decimals of the token used for the payment. */
+  decimal: number;
+}
+
+/**
+ * Enum for representing the supported sorting orders of the api.
+ *
+ * @public
+ */
+export enum SortingOrder {
```

Comment:
Yup, I think it would be a good idea to refactor it later to have one enum representing the supported sorting orders by the API. I didn't do it in this pr to avoid a breaking change with the `AssetTransfersOrder`